vrs/marten-construction-script.py

```diff
@@ -1,168 +1,180 @@
 
-# checkpoint for final multiallelic variants
-multiallelic_path = "gs://gnomad-tmp-4day/marten-temp-01-06/multiallelic.ht"
-ht_multiallelic = ht_multiallelic.checkpoint(multiallelic_path, overwrite = True)# , overwrite=overwrite, _read_if_exists=read_if_exists)
-# ^^^ TEMPORARY PATH
+# Checkpoint for final multiallelic variants
+ht_multiallelic = ht_multiallelic.checkpoint(f'{tmp_path}multiallelic_path_temp.ht', _read_if_exists=read_if_var, overwrite = overwrite_var)
 
 # Construct table of variants altered by min-rep
 
-# code in this section provided by Julia Goodrich (best boss this side of the Mississippi)
-vds = hl.vds.read_vds("gs://gnomad/v3.1/raw/gnomad_v3.1.vds")
-ht_minrep = vds.variant_data.rows()
+vds = VersionedVariantDatasetResource(
+    '3.1',
+    {"3.1": VariantDatasetResource("gs://gnomad/v3.1/raw/gnomad_v3.1.vds")},
+)
+
+ht_minrep = vds.vds().variant_data.rows()
 ht_minrep = ht_minrep.annotate(original_alleles=ht_minrep.alleles)
 ht_minrep = hl.split_multi(ht_minrep)
 ht_minrep = ht_minrep.filter(ht_minrep.alleles[1] != ht_minrep.original_alleles[ht_minrep.a_index])
-
-ht_minrep_filt = ht_minrep._filter_partitions(sample(range(ht_minrep.n_partitions()), 5))
-ht_minrep_filt = ht_minrep_filt.select()
-ht_minrep_final = ht_minrep_filt.join(ht_whole, how = "inner")
+ht_minrep = ht_minrep._filter_partitions(sample(range(ht_minrep.n_partitions()), 5))
+ht_minrep = ht_minrep.select()
+ht_minrep = ht_minrep.join(ht_whole, how = "inner")
+# an inner join here joins on Locus and Allele
+# doing the same using a collect() and filter() step would be awkward, since it involves two separate fields
+# doing the following 
 # ^ results in some few thousand variants, all being samples changed by min-rep
 
 # checkpoint for final minrep variants
-minrep_path_final = "gs://gnomad-tmp-4day/marten-temp-01-06/minrep-checkpoint.ht"
-ht_minrep_final = ht_minrep_final.checkpoint(minrep_path_final, overwrite = True)
-
-minrepcount = ht_minrep_final.count()
-print(f'There are {minrepcount} variants in the minrep set')
-
-# Final output and construction steps
+ht_minrep = ht_minrep.checkpoint(f'{tmp_path}minrep-checkpoint.ht', _read_if_exists=read_if_var, overwrite = overwrite_var)
 
+# Counts and further assembly
+minrepcount = ht_minrep.count()
+logger.info(f'There are %i variants in the minrep set', minrepcount)
 ht_prefinal = ht_union.union(ht_multiallelic)
-ht_final = ht_prefinal.union(ht_minrep_final)
-final_path = "gs://gnomad-tmp/marten-temp-01-06/final_output.ht"
-ht_final = ht_final.checkpoint(final_path,overwrite = True)
-# this also writes the table to where you want it, so you can set this as the final output!!!
-
-ht_final.show(5)
-ht_final.describe()
-ht_final.count()
-
-## spiked-in Extra ERRORS Coming Soon !!! 
+ht_final = ht_prefinal.union(ht_minrep)
+
+# Check for duplicates and de-duplication
+org_count = ht_final.count()
+ht_final = ht_final.distinct()
```

Comment:
adding a checkpoint here could help with the export (rather than adding a write at the end of the script (this line: `ht_final.write(final_path_hail)`)